packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -1445,18 +1451,19 @@ export class CacheDataClient implements IDataClient {
                 break;
               case grpcCache.ECacheResult.Hit: {
                 const shouldDecompress =
-                  options?.decompress ?? this.autoDecompressEnabled;
+                  options?.decompress ??
+                  this.compressionDetails?.autoDecompressEnabled === true;
```

Comment:
For my benefit, is the `=== true` necessary?

Reply:
it's a compiler warning/error without it, because the type would technically be `boolean | undefined`. This forces it to be a boolean.

Reply:
I don't think so, since `this.compressionDetails?.autoDecompressEnabled` will evaluate as false if it is nil. I don't think it should be able to be nil though, since automatic decompression is enabled if it isn't specified.